### PR TITLE
Battery

### PR DESCRIPTION
**Pull request by dguittet**
Add battery unit model and test file

part of Issue #14

Same as #12

### CODE REVIEW

**Comment by dguittet** [comment]
@jghouse88 Two questions:

1. Since time is handled externally to this model, do the variables being single values (as opposed to indexed by time domain) here make sense? 
2. Should the nameplate power and energy be variables or parameters?

File: dispatches/models/renewables_case/test_battery.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    m.fs.battery.initial_energy_throughput.fix(0)
+    m.fs.battery.nameplate_power.set_value(5)
+    m.fs.battery.nameplate_energy.fix(20)
+    m.fs.battery.elec_in.fix(5)
+    m.fs.battery.elec_out.fix(0)
+
+    assert_units_consistent(m)
+
+    solver = SolverFactory('ipopt')
+    solver.solve(m.fs)
```
Comment (andrewlee94) [inline dispatches/models/renewables_case/test_battery.py:53]: We should check solver status here as well.

File: dispatches/models/renewables_case/battery.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        """
+        super().build()
+
+        self.dt = Param(within=NonNegativeReals,
+                        initialize=1,
+                        doc="Time step",
+                        units=pyunits.hr)
+
+        self.nameplate_power = Var(within=NonNegativeReals,
+                                   initialize=0.0,
```
Comment (andrewlee94) [inline dispatches/models/renewables_case/battery.py:69]: We should consider adding bounds to these.

File: dispatches/models/renewables_case/battery.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        """Building model
+
+        Args:
+            None
+        Returns:
+            None
+        """
+        super().build()
+
+        self.dt = Param(within=NonNegativeReals,
```
Comment (andrewlee94) [inline dispatches/models/renewables_case/battery.py:63]: I would suggest more descriptive names for this, to help users in identifying what they mean/do.

Do we also need a comment describing why there is a separate `dt` from the `Flowsheet` time domain (and how the user should set this)?

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
A few comments. The one about checking solver status is probably the most important.

File: dispatches/models/renewables_case/battery.py (hunk 2 above)
Comment (dguittet) [reply dispatches/models/renewables_case/battery.py:69]: Okay, something like 1000 MW max power, and 10,000 MWh max capacity?

Comment (Andresj89) [reply dispatches/models/renewables_case/battery.py:68]: Those bounds should be good

**Review by Andresj89 — APPROVED** [review]
LGTM

**Review by jghouse88 — APPROVED** [review]
Looks good @dguittet

**Comment by jghouse88** [comment]
> 
> 
> @jghouse88 Two questions:
> 
>     1. Since time is handled externally to this model, do the variables being single values (as opposed to indexed by time domain) here make sense?
> 
>     2. Should the nameplate power and energy be variables or parameters?

1. Yes, that can be handled externally.
2. Yes and I think you had declared them as vars.

**Review by andrewlee94 — APPROVED** [review]
Looks good.

**Pull request merged**